### PR TITLE
Replace ambiguous Message struct with RetainedMessage(Batch)

### PR DESCRIPTION
**Pull request by numinnex**
Rename Message to PolledMessage and replace it's usage on the server with RetainedMessage and RetainedMessageBatch

### CODE REVIEW

**Review by hubcio — DISMISSED** [review]

File: README.md [hunk 1]
```diff
@@ -1,11 +1,10 @@
 # Iggy
-
-<div style="display: flex; flex-wrap: wrap; justify-content: center; align-items: center; text-align: center;">
-
+<div align="center">
```
Comment (hubcio) [inline README.md]: this change is not related to your pr

File: server/src/streaming/segments/index.rs [hunk 2]
```diff
@@ -24,3 +62,117 @@ impl IndexRange {
...
+        let mut segment = create_segment();
+        create_test_indices(&mut segment);
+        let result = segment
+            .load_highest_lower_bound_index(segment.indexes.as_ref().unwrap(), 15, 45)
+            .unwrap();
+
+        assert_eq!(result.start.relative_offset, 20);
+        assert_eq!(result.end.relative_offset, 50);
+    }
+    #[test]
```
Comment (hubcio) [inline server/src/streaming/segments/index.rs:143]: add empty line betwen `}` and `#[test]`

File: server/src/streaming/batching/message_batch.rs [hunk 3]
```diff
@@ -0,0 +1,166 @@
...
+    fn get_size_bytes(&self) -> u32 {
+        8 + 4 + 8 + 4 + self.length
+    }
+}
+
+impl Sizeable for Arc<RetainedMessageBatch> {
+    fn get_size_bytes(&self) -> u32 {
+        8 + 4 + 8 + 4 + self.length
+    }
+}
```
Comment (hubcio) [inline server/src/streaming/batching/message_batch.rs:101]: you can replace these two with
```rust
impl<T> Sizeable for T
where
    T: Deref<Target = RetainedMessageBatch>,
{
    fn get_size_bytes(&self) -> u32 {
        8 + 4 + 8 + 4 + self.length
    }
}
```

File: server/src/streaming/partitions/partition.rs [hunk 4]
```diff
@@ -136,6 +137,7 @@ impl Partition {
             config,
             storage,
             created_at: IggyTimestamp::now().to_micros(),
+            avg_timestamp_delta: 0,
```
Comment (hubcio) [inline server/src/streaming/partitions/partition.rs]: add unit OR change this type to IggyDuration

File: server/src/streaming/partitions/storage.rs [hunk 5]
```diff
@@ -221,7 +221,7 @@ impl Storage<Partition> for FilePartitionStorage {
             );
             segment.load().await?;
             if !segment.is_closed {
-                segment.unsaved_messages = Some(Vec::new())
+                segment.unsaved_batches = Some(Vec::new())
```
Comment (hubcio) [inline server/src/streaming/partitions/storage.rs:224]: consider using `Vc` from `atone` instead of `Vec`

Comment (numinnex) [reply server/src/streaming/partitions/storage.rs:224]: Going to do that in another pull request, proper struct wrapping unsaved_batches buffer is required in order to expose api, similar to Vec<T>

Comment (hubcio) [reply server/src/streaming/partitions/storage.rs:224]: ok

**Review by hubcio — APPROVED** [review]

**Review by spetz — APPROVED** [review]
